tests/core/test_utilities.py

```diff
@@ -913,6 +914,103 @@ def test_fit_plane_to_points():
     )
 
 
+# Default output from `np.linalg.eigh`
+DEFAULT_PRINCIPAL_AXES = [[0.0, 0.0, 1.0], [0.0, 1.0, 0.0], [-1.0, 0.0, 0.0]]
+
+CASE_RANK0 = (
+    0,
+    [[0, 0, 0], [0, 0, 0]],
+    [DEFAULT_PRINCIPAL_AXES],
+)
+CASE_RANK1 = (
+    1,
+    [[0, 0, 0], [1, 0, 0]],
+    [[1.0, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, -1.0, 0.0]],
+)
+
+CASE_RANK2 = (
+    2,
+    [[1, 2, 3], [3, 2, 1]],
+    [
+        [-0.70710678, 0.0, 0.70710678],
+        [0.0, 1.0, 0.0],
+        [-0.70710678, 0.0, -0.70710678],
+    ],
+)
+CASE_RANK3 = (
+    3,
+    pv.examples.load_airplane().points,
+    [
+        [8.1159601e-07, -9.9255711e-01, -1.2178054e-01],
+        [-1.0000000e00, -8.0911440e-07, -6.9828459e-08],
+        [-2.9225653e-08, 1.2178054e-01, -9.9255711e-01],
+    ],
+)
```

Comment:
> Do you know why/how rank was important in the original implementation?

Between the different implementations (NumPy SVD, NumPy PCA, and and VTK OBB), some of them fail for certain edge cases, e.g. one or two points,  many points but they're planar, etc. This required special handling for some of the implementations. So the main intent was to have coverage for these cases, especially if the implementation may change in the future.

> The ranks got me a bit confused here. I expected the two-point cases to always have rank 1, because two points always define a line.

> I think what's going on here is that we're checking the rank of the points array, which is a bit misleading. I think we should be checking the rank of the centered points array, in which case all 2 point cases will be rank1 unless the two points coincide, see the current rank 0 test case.

"Rank" is the term I settled on to describe the edge cases but maybe we can call it something else. I wasn't so much thinking about the actual meaning or definition of it but rather just "what value does `np.linalg.matrix_rank` return". Maybe let's just call them "TEST_CASES" generally instead? And forgo any matrix rank checks.

Reply:
OK, I think your "rank 1" and "rank 2" are the same edge case. I can't check now but I'd expect both to have 2 zero sizes, the long axis along `points[1] - points[0]`, and two other indeterminate axes in the perpendicular plane.

And you're missing the rank 2 case where you have 3 points: one zeeo size, two nonzero sizes, and the zero axis is perpendicular to the plane spanned by the three points.

Reply:
> As far as my semi-naive picture goes, I'd expect something like this:
> 
> * rank 0: e.g. what you have now, or a single point
> * rank 1: any 2 non-coincidental points, e.g. what you have now
> * rank 2: any 3 non-collinear points, e.g. `[[1, 0, 0], [0, 1, 0], [0, 0, 1]]`
> * rank 3: any 4 non-coplanar points, e.g. `[[1, 0, 0], [0, 1, 0], [0, 0, 1], [-1, -1, -1]]`

Tests are updated in https://github.com/pyvista/pyvista/pull/6346/commits/a521e83cae901c0f3bb9f06ab949773a3e0bff09

The rank 2 case suggested above has a `np.linalg.matrix_rank()` equal to 3, so using term rank may be confusing, so the cases are renamed and the rank assertion has been remove. The intent is communicated in comments instead. 
Plus, this rank 2 case spat out `nan` as its 3rd std value. Not sure if we like that, or should force it zero? This behaviour doesn't seem consistent with other cases, e.g. the single point cases doesn't have nans, but maybe I would have expected it to based on the rank2 behaviour.

Reply:
> * rank 3: any 4 non-coplanar points, e.g. `[[1, 0, 0], [0, 1, 0], [0, 0, 1], [-1, -1, -1]]`

The axes for this test in python 3.8 are:
```
[[ 0.57735027, 0.57735027, 0.57735027],
 [-0.44225427, -0.37327035, 0.81552462],
 [ 0.68635109, -0.72617898, 0.03982789]],
```
But it's expected to be:    
```
[[-0.57735027, -0.57735027, -0.57735027],
 [0.0, -0.70710678, 0.70710678],
 [-0.81649658, 0.40824829, 0.40824829]]
```
The axes are still a valid result, but this makes it hard annoying to test since we need to special-case it. So a skip is added in https://github.com/pyvista/pyvista/pull/6346/commits/4b807de1d0b44382c4b2f675986d699426ca58da